service/storage_proxy.cc

```diff
@@ -2770,6 +2771,8 @@ class abstract_read_resolver {
         } catch (rpc::timeout_error&) {
             // do not report timeouts, the whole operation will timeout and be reported
             return; // also do not report timeout as replica failure for the same reason
+        } catch (const exceptions::overloaded_exception& e) {
```

Comment:
Exception type is not propagated over rpc, so this will work only for local failures.

Reply:
That's what I was afraid of... is it not propagated at all because there's no place for it in the current protocol, or do we have some way to send different error codes? Could you point me to the code which translates the replica-side error to something sent via RPC? I couldn't find when authoring this rfc.

Reply:
You can send error code as part of normal response (an rpc verb can be defined to return variant<error, response>, but all exceptions (except rpc internal one) are propagated as "an exception with _this_ error string". The relevant code is in RPC. You can search for unmarshal_exception() to find where to dig further.